'class esphome::pvvx_mithermometer::PVVXDisplay' has no member named 'time_'

### The problem

Hi,

I'm attempting to integrate a Xiaomi LYWSD03MMC running PVVX's ATC_MiThermometer firmware v4.5 as a remote display for my Home Assistant setup.
I've started with the first example on the [component's page](https://esphome.io/components/display/pvvx_mithermometer.html) but the build fails on a clean install of HAOS running the ESPHome add-on, with the build files cleared before each build attempt.

It appears that the lack of a 'time' component is not gracefully handled by the pvvx component.
As such, I would recommend adding a dependency notice, as per example [Display states of other sensors](https://esphome.io/components/display/pvvx_mithermometer.html#display-states-of-other-sensors) so that other may no fall in this trap.
```
time:
  - platform: homeassistant
    id: homeassistant_time
[...]
display:
- platform: pvvx_mithermometer
  time_id: homeassistant_time

### Which version of ESPHome has the issue?

2023.11.6

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2023.11.3

### What platform are you using?

ESP32

### Board

esp32-s3-devkitc-1

### Component causing the issue

pvvx_mithermometer

### Example YAML snippet

```yaml
esp32_ble_tracker:

ble_client:
  - mac_address: A4:C1:XX:XX:XX:XX
    id: displ

display:
- platform: pvvx_mithermometer
  ble_client_id: displ
  lambda: |-
    it.print_bignum(23.1);
    it.print_unit(pvvx_mithermometer::UNIT_DEG_C);
    it.print_smallnum(33);
    it.print_percent(true);
    it.print_happy(true);
    it.print_bracket(true);
```


### Anything in the logs that might be useful for us?

```txt
In file included from src/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp:2:
src/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp: In member function 'virtual void esphome::pvvx_mithermometer::PVVXDisplay::dump_config()':
src/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp:16:66: error: 'class esphome::pvvx_mithermometer::PVVXDisplay' has no member named 'time_'
   ESP_LOGCONFIG(TAG, "  Set time on connection: %s", YESNO(this->time_ != nullptr));
                                                                  ^~~~~
src/esphome/core/log.h:97:90: note: in definition of macro 'esph_log_config'
   esp_log_printf_(ESPHOME_LOG_LEVEL_CONFIG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
                                                                                          ^~~~~~~~~~~
src/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp:16:3: note: in expansion of macro 'ESP_LOGCONFIG'
   ESP_LOGCONFIG(TAG, "  Set time on connection: %s", YESNO(this->time_ != nullptr));
   ^~~~~~~~~~~~~
src/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp:16:54: note: in expansion of macro 'YESNO'
   ESP_LOGCONFIG(TAG, "  Set time on connection: %s", YESNO(this->time_ != nullptr));
                                                      ^~~~~
Compiling .pioenvs/esphome-web-0d7f88/src/esphome/components/sensor/automation.cpp.o
Compiling .pioenvs/esphome-web-0d7f88/src/esphome/components/sensor/filter.cpp.o
*** [.pioenvs/esphome-web-0d7f88/src/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp.o] Error 1
```


### Additional information

_No response_

--- a/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp
+++ b/esphome/components/pvvx_mithermometer/display/pvvx_display.cpp
@@ -13,7 +13,9 @@ void PVVXDisplay::dump_config() {
   ESP_LOGCONFIG(TAG, "  Service UUID          : %s", this->service_uuid_.to_string().c_str());
   ESP_LOGCONFIG(TAG, "  Characteristic UUID   : %s", this->char_uuid_.to_string().c_str());
   ESP_LOGCONFIG(TAG, "  Auto clear            : %s", YESNO(this->auto_clear_enabled_));
+#ifdef USE_TIME
   ESP_LOGCONFIG(TAG, "  Set time on connection: %s", YESNO(this->time_ != nullptr));
+#endif
   ESP_LOGCONFIG(TAG, "  Disconnect delay      : %" PRIu32 "ms", this->disconnect_delay_ms_);
   LOG_UPDATE_INTERVAL(this);
 }

